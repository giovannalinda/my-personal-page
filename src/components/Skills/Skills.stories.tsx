import { Meta, Story } from '@storybook/react'
import { Skills } from './Skills'

export default {
  title: 'Components/Skills',
  component: Skills,
} as Meta

export const Default: Story = () => <Skills />
