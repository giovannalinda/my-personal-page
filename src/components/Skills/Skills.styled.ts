import { motion } from 'framer-motion'
import styled from 'styled-components'
import { theme } from '~/styles'

export const List = styled(motion.ul)`
  display: grid;
  grid-template-columns: repeat(5, 1fr);
  text-align: center;
  justify-content: center;
  gap: 1rem;
  margin-bottom: 30rem;

  @media screen and (max-width: ${theme.breakpoints.smallDesktop}) {
    grid-template-columns: repeat(2, 1fr);
    padding: ${theme.space[7]};
  }
`

export const Title = styled.h1`
  font-size: ${theme.font.sizes.subtitle};
  font-weight: ${theme.font.weigths.regular};
  padding-bottom: ${theme.space[7]};

  @media screen and (max-width: ${theme.breakpoints.smallDesktop}) {
    padding-left: ${theme.space[7]};
  }
`

export const ListItem = styled(motion.li)`
  padding: ${theme.space[7]};
  border: 1px solid ${theme.colors.gray800};
  font-size: ${theme.font.sizes.small};
  transition: ${theme.transitions.slow};

  &:hover {
    border-color: ${theme.colors.white};
  }
`

export const Paragraph = styled.p`
  padding-top: ${theme.space[4]};
`
