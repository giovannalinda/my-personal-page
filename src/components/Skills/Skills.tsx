import {
  SiHtml5,
  SiCss3,
  SiJavascript,
  SiReact,
  SiTypescript,
  SiNextdotjs,
  SiStyledcomponents,
  SiStorybook,
  SiJest,
  SiFigma,
} from 'react-icons/si'

import { InView } from 'react-intersection-observer'

import { useTranslation } from '~/lib'

import * as S from './Skills.styled'

const skills = [
  {
    name: 'HTML',
    logo: <SiHtml5 size={80} />,
  },
  {
    name: 'CSS',
    logo: <SiCss3 size={80} />,
  },
  {
    name: 'JavaScript',
    logo: <SiJavascript size={80} />,
  },
  {
    name: 'React',
    logo: <SiReact size={80} />,
  },
  {
    name: 'TypeScript',
    logo: <SiTypescript size={80} />,
  },
  {
    name: 'Next',
    logo: <SiNextdotjs size={80} />,
  },
  {
    name: 'Styled Components',
    logo: <SiStyledcomponents size={80} />,
  },
  {
    name: 'Storybook',
    logo: <SiStorybook size={80} />,
  },
  {
    name: 'Jest',
    logo: <SiJest size={80} />,
  },
  {
    name: 'Figma',
    logo: <SiFigma size={80} />,
  },
]

const cardVariant = {
  hidden: {
    opacity: 0,
  },
  visible: {
    opacity: 1,
  },
}

export function Skills() {
  const { t } = useTranslation()

  return (
    <InView triggerOnce>
      {({ inView, ref }) => (
        <div ref={ref}>
          <S.Title>{t('Skills')}</S.Title>

          {inView && (
            <S.List
              initial='hidden'
              animate='visible'
              variants={cardVariant}
              transition={{
                duration: 0.1,
                delay: 1,
              }}
            >
              {skills.map((skill, index) => (
                <S.ListItem
                  key={skill.name}
                  aria-label={skill.name}
                  initial='hidden'
                  animate='visible'
                  variants={cardVariant}
                  transition={{
                    duration: 0.5,
                    delay: index / skills.length,
                  }}
                >
                  {skill.logo}
                  <S.Paragraph>{skill.name}</S.Paragraph>
                </S.ListItem>
              ))}
            </S.List>
          )}
        </div>
      )}
    </InView>
  )
}
