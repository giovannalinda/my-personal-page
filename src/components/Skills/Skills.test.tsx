import { render } from '~/__helpers__/app-tests'
import { Skills } from './Skills'

describe('Skills', () => {
  it('matches snapshot', () => {
    const { container } = render(<Skills />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
