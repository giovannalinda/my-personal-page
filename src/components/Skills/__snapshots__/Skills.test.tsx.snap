// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills matches snapshot 1`] = `
<h1
  class="sc-gsnTZi dSgLBd"
>
  Conhecimentos
</h1>
`;
